app/assets/javascripts/discourse/controllers/ignore-duration.js.es6

```diff
@@ -17,6 +18,8 @@ export default Ember.Controller.extend(ModalFunctionality, {
       this.get("model")
         .updateNotificationLevel("ignore", this.get("ignoredUntil"))
         .then(() => {
+          const currentUser = PreloadStore.get("currentUser");
+          currentUser.ignored_users.push(this.get("model.username"));
```

Comment:
I could probably avoid this logic entirely by exposing some method of refreshing the preloaded `currentUser`.
The technique we use the [findDetails](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/models/user.js.es6#L469-L517) here sounds similar to what I want to achieve since it also `refreshes` the `preloaded-store`.